react/features/base/tracks/functions.js

```diff
@@ -159,20 +159,35 @@ export function getTracksByMediaType(tracks, mediaType) {
 /**
  * Mutes or unmutes a specific <tt>JitsiLocalTrack</tt>. If the muted state of
  * the specified <tt>track</tt> is already in accord with the specified
- * <tt>muted</tt> value, then does nothing.
+ * <tt>muted</tt> value, then does nothing. In case mute/unmute operation fails
+ * (JitsiLocalTrack Promise is rejected) a rollback action will be dispatched on
+ * the given store. For example if the mute operation fails then the 'unmute'
+ * action will be dispatched to rollback to the previous state in base/media.
  *
  * @param {JitsiLocalTrack} track - The <tt>JitsiLocalTrack</tt> to mute or
  * unmute.
  * @param {boolean} muted - If the specified <tt>track</tt> is to be muted, then
  * <tt>true</tt>; otherwise, <tt>false</tt>.
- * @returns {Promise}
+ * @param {Store} store - The redux store in the context of which the function
+ * is to execute and which will be used to dispatch the rollback action in case
+ * mute/unmute fails.
+* @returns {Promise}
  */
-export function setTrackMuted(track, muted) {
+export function setTrackMuted(track, muted, { dispatch }) {
```

Comment:
Thank you very much, @paweldomas & @saghul!

I'd like to point out for the future that a function which `dispatch`es an action in the store and, optionally, performs some additional work before and/or after the `dispatch` is an asychronous action creator. So in this case `setTrackMuted` is a very good candidate for `actions.js`. I'll move it there this time.